component/src/main/java/org/wso2/extension/siddhi/execution/env/ResourceBatchWindowProcessor.java

```diff
@@ -0,0 +1,279 @@
+ * under the License.
+ */
+package org.wso2.extension.siddhi.execution.env;
+
+import org.wso2.siddhi.annotation.Example;
+import org.wso2.siddhi.annotation.Extension;
+import org.wso2.siddhi.annotation.Parameter;
+import org.wso2.siddhi.annotation.util.DataType;
+import org.wso2.siddhi.core.config.SiddhiAppContext;
+import org.wso2.siddhi.core.event.ComplexEvent;
+import org.wso2.siddhi.core.event.ComplexEventChunk;
+import org.wso2.siddhi.core.event.stream.StreamEvent;
+import org.wso2.siddhi.core.event.stream.StreamEventCloner;
+import org.wso2.siddhi.core.executor.ConstantExpressionExecutor;
+import org.wso2.siddhi.core.executor.ExpressionExecutor;
+import org.wso2.siddhi.core.query.processor.Processor;
+import org.wso2.siddhi.core.query.processor.SchedulingProcessor;
+import org.wso2.siddhi.core.query.processor.stream.window.WindowProcessor;
+import org.wso2.siddhi.core.util.Scheduler;
+import org.wso2.siddhi.core.util.config.ConfigReader;
+import org.wso2.siddhi.query.api.definition.Attribute;
+import org.wso2.siddhi.query.api.exception.SiddhiAppValidationException;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Implementation of {@link WindowProcessor} which represent a Resource Batch Window operating based on
+ * the inferred resources count from env:resourceIdentify extension as length.
+ */
+@Extension(
+        name = "resourceBatch",
+        namespace = "env",
+        description = "A resource batch (tumbling) window that holds a number of events with specified attribute " +
+                "as grouping key and based on the resource count inferred from env:resourceIdentify extension. " +
+                "The window is updated each time a batch of events with same key value that equals the number of " +
+                "resources count.",
+        parameters = {
+                @Parameter(name = "resource.name",
+                        description = "The resource name.",
+                        type = {DataType.STRING}),
+                @Parameter(name = "group.key.name",
+                        description = "The attribute that should be used for events grouping.",
+                        type = {DataType.INT, DataType.LONG, DataType.FLOAT,
+                                DataType.BOOL, DataType.DOUBLE}),
+                @Parameter(name = "time.in.milliseconds",
+                        description = "Time to wait for arrival of new event, before flushing " +
+                                "and giving output for events belonging to a specific batch.",
+                        type = {DataType.INT, DataType.LONG, DataType.TIME},
+                        optional = true,
+                        defaultValue = "300000")
+        },
+        examples = {
+                @Example(
+                        syntax = "define window cseEventWindow (symbol string, price float, volume int) " +
```

Comment:
Have an example that has both the extensions with a meaningful use-case

Reply:
Remove the 1st example does not make much sense 
Also use first later capital when you define Streams/Windows etc